UPGRADE-1.14.md

```diff
@@ -0,0 +1,23 @@
+# UPGRADE FROM 1.x to 1.14
+
+## Method signature changes
```

Comment:
> I've skipped a list of changes for this PR, but can list method signature changes in the upgrade document as well. An example of this is the [upgrade document for doctrine/collections](https://github.com/doctrine/collections/blob/2.0.x/UPGRADE.md#doctrinecommoncollectionscollection) where interfaces were changed to add types.

That format looks good to me.

> We'll want to consider where to document upcoming return type changes for ext-mongodb. While users on PHP 8.1 and newer will be informed of tentative return types, users on previous version will not be aware of those changes.

Two thoughts here. Does it make sense to document all signature changes in the PHPC method changelogs? That seems like the expected place for it and would certainly be more accessible than a separate page in the PHP.net docs that might be easily overlooked.

I also wouldn't object to including PHPC API changes in this document, since those are going to arrive in the corresponding PHPC release (1.15 for PHPLIB 1.14).

Reply:
> Does it make sense to document all signature changes in the PHPC method changelogs?

Yes. I've already added a note about tentative return types (see https://github.com/php/doc-en/commit/734bafeaf071b78b15d375f9af583befddd8c2a2), but calling out other changes (such as widening parameter types, etc.) should also be called out. I'll go through the changes and make sure that those are updated.

There aren't any other significant API changes for users, as we've only changed final classes in PHPC. However, it will become relevant for 2.0, at which point I'd definitely point out PHPC API changes in this upgrade document as well where it affects users.